### DISCUSSION TITLE
How can I understand which paths are taking long when "Enumeration of workspace source files is taking longer than 10 seconds."?

### DISCUSSION THREAD

**Discussion by elongl** [microsoft/pyright#8003] [Q&A]
I'm opening a repository in VSCode and indexing the source code takes a while.
I think this happens because Pyright enumerates files that it shouldn't, but I can't figure out which so I can exclude them.
I've also set the log-level to `Trace`, but there are no logs on which files it is enumerating.
How can I solve this?
Thanks.

**Comment by erictraut** [comment]
Are you using pyright or pylance? Pyright doesn't include any indexing functionality. If you suspect that there's a problem with pylance's indexing functionality, please log a bug (or post a question) in the [pylance-release](https://github.com/microsoft/pylance-release) project.

Pyright does enumerate the files in your project when you open the project's folder. It does this by starting at the root directory and recursing through subdirectories (including symbolic links). It stops recursing if it encounters a directory that matches one of the globs specified in the "exclude" configuration setting.

To determine which root directory pyright is using as the start of its enumeration, you can look at the log output. Search for a line that starts with `Server root directory: `.

**Comment by qsunyuan** [comment]
I have also encountered this problem. There is a large amount of code / folders (even maybe some data list) in my workspace, which causes pylance to not work. This has cost me a lot of time. Here may be a feasible solution: I manually traverse and overlook some paths to find the corresponding folder.

`pyrightconfig.json`
```json
{ 
    "exclude":[ 
      "folder1",   "folder2",   
    ], 
}
```
